sources/whippet/src/child.rs

```diff
@@ -0,0 +1,128 @@
+    pub(crate) fn new(broker_path: &str) -> Result<Self> {
+        Ok(Self {
+            broker_path: broker_path.to_string(),
+        })
+    }
+
+    /// Forks the dbus-broker with the parameters configured for the child
+    pub(crate) fn run(
+        self,
+        journal_socket: UnixDatagram,
+        broker_socket: UnixStream,
+    ) -> Result<BrokerChildRunning> {
+        debug!("Reading machine ID from {MACHINE_ID_PATH}");
+        let machine_id = std::fs::read_to_string(MACHINE_ID_PATH)
+            .context(error::MachineIdSnafu {
+                path: MACHINE_ID_PATH,
+            })?
+            .trim()
+            .to_string();
+        debug!("Machine ID: {machine_id}");
+
+        let mut cmd = Command::new(&self.broker_path);
+        cmd.arg("--log")
+            .arg(format!("{}", journal_socket.as_raw_fd()))
+            .arg("--controller")
+            .arg(format!("{}", broker_socket.as_raw_fd()))
+            .arg("--machine-id")
+            .arg(machine_id)
+            .arg("--max-bytes")
+            .arg(format!("{MAX_BYTES}"))
+            .arg("--max-fds")
+            .arg(format!("{MAX_FDS}"))
+            .arg("--max-matches")
+            .arg(format!("{MAX_MATCHES}"))
+            .stdin(Stdio::inherit())
+            .stdout(Stdio::inherit())
+            .stderr(Stdio::inherit());
+
+        debug!(
+            "Spawning dbus-broker with limits: max-bytes={MAX_BYTES}, max-fds={MAX_FDS}, max-matches={MAX_MATCHES}",
+        );
+
+        // Hold the reference to the spawned process, so that it can be killed if needed
+        let child = cmd.spawn().context(error::BrokerSpawnSnafu)?;
+        info!("dbus-broker spawned with PID: {:?}", child.id());
+
+        // Pass down the sockets to the active running child, to prevent them from going out of
+        // scope before the child exits
+        Ok(BrokerChildRunning {
+            child,
+            _journal_socket: journal_socket,
+            _broker_socket: broker_socket,
+        })
+    }
+}
+
+impl BrokerChildRunning {
+    /// Wait for either the spawned process to complete, or for the SIGTERM signal to be sent
+    pub(crate) async fn wait(mut self) -> Result<()> {
+        debug!("Setting up signal handlers");
+        let mut sigterm = signal::unix::signal(signal::unix::SignalKind::terminate())
+            .context(error::CreateSigTermSignalSnafu)?;
+        // Wait for either the child process to stop, or for the kill signal
+        tokio::select! {
+            // Child process died naturally
+            status = self.child.wait() => {
+                let status = status.context(error::BrokerWaitSnafu)?;
+                if status.success() {
+                    info!("dbus-broker exited successfully: {status}");
+                } else {
+                    warn!("dbus-broker exited with status: {status}");
+                }
+                Ok(())
+            }
+            _ = sigterm.recv() => {
```

Comment:
Do we care about SIGINT, can anything send other signals besides SIGTERM?

Reply:
SIGINT is handled by `ctrl_c` above. And we don't care about other signals. The other signal `whippet` might care about in the future is `HUP` but only if we ever decide to support reloading the service without bringing the broker down.